File: java-codellama7B-AWQ-all/taskid-156_int_to_mini_roman-gen13.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    // Given a positive integer, obtain its roman numeral equivalent as a string,
    // and return it in lowercase.
    // Restrictions: 1 <= num <= 1000
    // Examples:
    // >>> intToMiniRoman((19l))
    // ("xix")
    // >>> intToMiniRoman((152l))
    // ("clii")
    // >>> intToMiniRoman((426l))
    // ("cdxxvi")
    public static String intToMiniRoman(long number) {

        if (number > 1000l) return "";
        if (number <= 10l) {
            return String.valueOf(number);
        }
        int thousands = (int)(number / 1000l);
        number = number % 1000l;
        int hundreds = (int)(number / 100l);
        number = number % 100l;
        String res = "";
        if (thousands > 0) {
            res += String.valueOf(thousands).concat("m");
        }
        if (hundreds > 0) {
            res += String.valueOf(hundreds).concat("c");
        }
        if (number >= 10l && number <= 100l) {
            res += String.valueOf(number).substring(0, 1);
        }
        if (number >= 1l && number < 10l) {
            res += String.valueOf(number);
        }
        return res;
    }
    public static void main(String[] args) {
    assert(intToMiniRoman((19l)).equals(("xix")));
    assert(intToMiniRoman((152l)).equals(("clii")));
    assert(intToMiniRoman((251l)).equals(("ccli")));
    assert(intToMiniRoman((426l)).equals(("cdxxvi")));
    assert(intToMiniRoman((500l)).equals(("d")));
    assert(intToMiniRoman((1l)).equals(("i")));
    assert(intToMiniRoman((4l)).equals(("iv")));
    assert(intToMiniRoman((43l)).equals(("xliii")));
    assert(intToMiniRoman((90l)).equals(("xc")));
    assert(intToMiniRoman((94l)).equals(("xciv")));
    assert(intToMiniRoman((532l)).equals(("dxxxii")));
    assert(intToMiniRoman((900l)).equals(("cm")));
    assert(intToMiniRoman((994l)).equals(("cmxciv")));
    assert(intToMiniRoman((1000l)).equals(("m")));
    }

}

File: py-codellama13B-FP-all/taskid-156_int_to_mini_roman-gen13.py
def int_to_mini_roman(number: int) -> str:
    """
    Given a positive integer, obtain its roman numeral equivalent as a string,
    and return it in lowercase.
    Restrictions: 1 <= num <= 1000

    Examples:
    >>> int_to_mini_roman(19)
    'xix'
    >>> int_to_mini_roman(152)
    'clii'
    >>> int_to_mini_roman(426)
    'cdxxvi'
    """

    roman_numeral = ''
    tens, digits = divmod(number, 10)
    if tens == 9:
        roman_numeral += 'xc'
    elif tens == 8:
        roman_numeral += 'lxxx'
    elif tens == 7:
        roman_numeral += 'lxx'
    elif tens == 6:
        roman_numeral += 'lx'
    elif tens == 5:
        roman_numeral += 'l'
    elif tens == 4:
        roman_numeral += 'xl'
    elif tens == 3:
        roman_numeral += 'xxx'
    elif tens == 2:
        roman_numeral += 'xx'
    elif tens == 1:
        roman_numeral += 'x'

    if digits == 9:
        roman_numeral += 'ix'
    elif digits == 8:
        roman_numeral += 'viii'
    elif digits == 7:
        roman_numeral += 'vii'
    elif digits == 6:
        roman_numeral += 'vi'
    elif digits == 5:
        roman_numeral += 'v'
    elif digits == 4:
        roman_numeral += 'iv'
    elif digits == 3:
        roman_numeral += 'iii'
    elif digits == 2:
        roman_numeral += 'ii'
    elif digits == 1:
        roman_numeral += 'i'

    return roman_numeral.lower()
